modal/cli/import_refs.py

```diff
@@ -31,51 +32,66 @@
 @dataclasses.dataclass
 class ImportRef:
     file_or_module: str
+    is_module: bool
 
     # object_path is a .-delimited path to the object to execute, or a parent from which to infer the object
     # e.g.
     # function or local_entrypoint in module scope
     # app in module scope [+ method name]
     # app [+ function/entrypoint on that app]
-    object_path: str
+    object_path: str = dataclasses.field(default="")
 
 
-def parse_import_ref(object_ref: str) -> ImportRef:
+def parse_import_ref(object_ref: str, is_module: bool = False) -> ImportRef:
     if object_ref.find("::") > 1:
         file_or_module, object_path = object_ref.split("::", 1)
     elif object_ref.find(":") > 1:
         raise InvalidError(f"Invalid object reference: {object_ref}. Did you mean '::' instead of ':'?")
     else:
         file_or_module, object_path = object_ref, ""
 
-    return ImportRef(file_or_module, object_path)
+    return ImportRef(file_or_module, is_module, object_path)
 
 
 DEFAULT_APP_NAME = "app"
 
 
-def import_file_or_module(file_or_module: str):
+def import_file_or_module(import_ref: ImportRef, base_cmd: str = ""):
```

Comment:
Should we be introducing an error here (or somewhere else?) for when you do `modal run -m foo.py` where `foo.py` is a local script, and `foo.py` is not a path to a module? Or will that trigger an error later on?

Reply:
That would trigger an error later when we try to import the module since is_module=True

Reply:
Not sure what you mean - all of the module/script validation is in the `import_file_or_module` function here.

"Validation" of the import itself being ok (file existing, having valid python syntax etc) is delegated to importlib.import_module which is also called in this function, so it should all be contained here afaik?

Reply:
I meant checking all the various ways that the spelling of the import ref and the use or non-use of the `-m` flag can disagree. But maybe you can't determine that a path without `.py` isn't a valid module until you actually try to import it?

Reply:
Oh maybe the problem is that i just didn't read far enough down to understand when "trigger an error later" actually applies 😁